scripts/pipeline/orthology_benchmark.py

```diff
@@ -17,18 +17,134 @@
+
+    if len(dump_cores) == 0:
+        raise RuntimeError(f"No cores found for the species set '{species_set_name}' on the specified host.")
+
+    dumps_dir = os.path.join(out_dir, species_set_name)
+    os.mkdir(dumps_dir)
+
+    script = os.path.join(os.environ["ENSEMBL_ROOT_DIR"], "ensembl-compara", "scripts", "dumps",
+                          "dump_gene_set_from_core.pl")
+
+    for core in dump_cores:
+        out_file = os.path.join(dumps_dir, core + ".fasta")
+        subprocess.run([script, "-core-db", core, "-host", host, "-port", str(port),
+                       "-outfile", out_file], capture_output=True, check=True)
+
+
+def find_latest_core(core_names: List[str]) -> str:
+    """Returns the latest core database among the available ones.
+
+    Args:
+        core_names: A list of cores for a species of interest.
+
+    Raises:
+        RuntimeError: If `core_names` is empty.
+
+    Returns:
+        Name of the latest core database (the latest Ensembl release, the latest version).
+
+    """
+    if len(core_names) == 0:
+        raise RuntimeError("Empty list of core databases. Cannot determine the latest one.")
+
+    rel_ver = [name.split("_core_")[1].split("_") for name in core_names]
+    rel_ver_int = [list(map(int, i)) for i in rel_ver]
+    rel_ver_arr = numpy.array(rel_ver_int)
+
+    n_cols = rel_ver_arr.shape[1]
+    i = 0
+    while i < n_cols:
+        # Find max value in the i-th column
+        # For the next iteration (i+1) consider only rows where i-th column == max value
+        max_col = numpy.amax(rel_ver_arr[:, i])
+        rows_ind = numpy.where(rel_ver_arr[:, i] == max_col)[0]
+        rel_ver_arr = rel_ver_arr[[rows_ind], :][0]
+        i += 1
+
+    latest_rel_ver = '_'.join(map(str, rel_ver_arr[0]))
+    core_name = [core for core in core_names if latest_rel_ver in core][0]
+
+    return core_name
+
+
+def get_core_names(species_names: List[str], host: str, port:int, user: str) -> Dict[str, str]:
+    """Returns the latest core database names for a list of species.
+
+    Args:
+        species_names: Species (genome) names.
+        host: Host for core databases.
+        port: Host port.
+        user: Server username.
+
+    Raises:
+        RuntimeError: If `species_list` is empty.
+        sqlalchemy.exc.OperationalError: If `user` cannot read from `host:port`.
+
+    Returns:
+        Dictionary mapping species (genome) names to the latest version of available core names.
+
+    """
+    if len(species_names) == 0:
+        raise RuntimeError("Empty list of species names. Cannot search for core databases.")
+
+    core_names = {}
+
+    eng = create_engine("mysql://" + user + "@" + host + ":" + str(port) + "/")
+    out_tmp = eng.execute("SHOW DATABASES LIKE '%%_core_%%'").fetchall()
+    all_cores = [i[0] for i in out_tmp]
+
+    for species in species_names:
+        core_name = [core for core in all_cores if species + "_core_" in core]
```

Comment:
Consider using f-string and `startswith` method:
```suggestion
        core_name = [core for core in all_cores if core.startswith(f"{species}_core_")]
```